plugins/modules/system/solaris_zone.py

```diff
@@ -193,7 +193,7 @@ def configure(self):
             self.module.fail_json(msg='Missing required argument: path')
 
         if not self.module.check_mode:
-            t = tempfile.NamedTemporaryFile(delete=False)
+            t = tempfile.NamedTemporaryFile(delete=False, mode='wt')
```

Comment:
Why not:
```suggestion
            t = tempfile.NamedTemporaryFile(delete=False, mode='w+t')
```
? That would be closer to the default value of `w+b` (https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile).

Reply:
The "+" indicates, it's opened for reading and writing, which is not the case here. It only needs to be opened for writing. which is the correct option IMO.

Reply:
I agree with @pmarkham, the module is creating a file that's consumed by `zoneadm -f` and thus you don't need to read on the handle; but I'll change to `w+t` if that's what's required for merge.

Reply:
I'm happy with the explanation, I just wanted to ask whether this is intentional or not :) No need to change it.